src/Elastic.Apm/Api/Service.cs

```diff
@@ -14,22 +19,122 @@ public class Service
 		public Framework Framework { get; set; }
 		public Language Language { get; set; }
 		public string Name { get; set; }
+		public Runtime Runtime { get; set; }
 
 		public override string ToString() => new ToStringBuilder(nameof(Service))
 		{
 			{ "Name", Name }, { "Agent", Agent }, { "Framework", Framework }, { "Language", Language },
 		}.ToString();
 
-		internal static Service GetDefaultService(IConfigurationReader configurationReader)
-			=> new Service
+		internal static Service GetDefaultService(IConfigurationReader configurationReader, IApmLogger loggerArg)
+		{
+			IApmLogger logger = loggerArg.Scoped(nameof(Service));
+			return new Service
 			{
 				Name = configurationReader.ServiceName,
 				Agent = new AgentC
 				{
 					Name = Consts.AgentName,
 					Version = typeof(Agent).Assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>().InformationalVersion
-				}
+				},
+				Runtime = GetRuntime(logger)
 			};
+		}
+
+		private static Runtime GetRuntime(IApmLogger logger)
```

Comment:
Not so important, but this has quite some internal logic, but it's in the `Api` namespace. We have e.g. the `Helpers` folder, I feel that it'd be a better place for this code. If you don't wanna move it, feel free to leave it here, this is just my 2 cents.

Reply:
I agree it's better be in some internal helper - I moved it there.